### ISSUE TITLE
useless_if_let_seq should not warn for composite ifs

### ISSUE THREAD

**Issue by Manishearth** [rust-lang/rust-clippy#1089]
It warns on:

``` rust
    let mut bubble_inline_sizes_separately = debug_options.bubble_widths;
    if debug_options.trace_layout {
        paint_threads = 1;
        layout_threads = 1;
        bubble_inline_sizes_separately = true;
    }
```

While this could be rewritten, I'm not sure if it should.

Additionally, we shouldn't use span_suggestion here since the suggestion isn't valid rust code.

**Comment by GnomedDev** [comment]
`useless_if_let_seq` does not exist anymore, and the closest name `useless_let_if_seq` doesn't trigger for ```rust
#![allow(unused_assignments, unused_variables)]
#![warn(clippy::useless_if_let_seq)]

pub fn example(bubble_widths: bool, trace_layout: bool) {
    let paint_threads;
    let layout_threads;

    let mut bubble_inline_sizes_separately = bubble_widths;
    if trace_layout {
        paint_threads = 1;
        layout_threads = 1;
        bubble_inline_sizes_separately = true;
    }
}
```

Should this be closed?
